File: src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/HomeActivityResponse.java

package io.zeitwert.fm.app.adapter.api.rest.dto;

import io.zeitwert.ddd.enums.adapter.api.jsonapi.dto.EnumeratedDto;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class HomeActivityResponse {

	private EnumeratedDto item; // rating, task
	private EnumeratedDto relatedTo; // rating: building, task: obj
	private EnumeratedDto owner;

	private EnumeratedDto user;
	private String dueAt;

	private String subject;
	private String content;
	private EnumeratedDto priority;

}

File: fm-domain/src/main/java/io/zeitwert/app/api/rest/dto/HomeActivityResponse.kt
package io.zeitwert.app.api.rest.dto

import io.zeitwert.app.api.jsonapi.EnumeratedDto

data class HomeActivityResponse(
	val item: EnumeratedDto, // rating, task
	val relatedTo: EnumeratedDto, // rating: building, task: obj
	val owner: EnumeratedDto? = null,
	val user: EnumeratedDto? = null,
	val dueAt: String? = null,
	val subject: String? = null,
	val content: String? = null,
	val priority: EnumeratedDto? = null,
)
